x-pack/metricbeat/module/azure/azure.go

```diff
@@ -126,9 +127,14 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) error {
 		return nil
 	}
 	// retrieve metrics
-	err = m.Client.GetMetricValues(report)
-	if err != nil {
-		return err
+	groupedMetrics := groupMetricsByResource(m.Client.Resources.Metrics)
+
+	for _, metrics := range groupedMetrics {
+		results := m.Client.GetMetricValues(metrics, report)
+		err := EventsMapping(results, m.BaseMetricSet.Name(), report)
+		if err != nil {
+			return err
```

Comment:
nit: errors.Wrap(err, "error running EventsMapping")

Reply:
added the suggestion